receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -5,28 +5,62 @@ package prometheusremotewritereceiver // import "github.com/open-telemetry/opent
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componentstatus"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/receiver"
-	"go.uber.org/zap"
 )
 
-func newRemoteWriteReceiver(_ receiver.Settings, _ *Config, _ consumer.Metrics) (receiver.Metrics, error) {
-	panic("need implement")
+func newRemoteWriteReceiver(settings receiver.Settings, cfg *Config, nextConsumer consumer.Metrics) (receiver.Metrics, error) {
+	return &prometheusRemoteWriteReceiver{
+		settings:     settings,
+		nextConsumer: nextConsumer,
+		config:       cfg,
+		server: &http.Server{
+			ReadTimeout: 60 * time.Second,
+		},
+	}, nil
 }
 
-// nolint
 type prometheusRemoteWriteReceiver struct {
-	settings     receiver.Settings //nolint:unused
-	host         component.Host    //nolint:unused
-	nextConsumer consumer.Metrics  //nolint:unused
+	settings     receiver.Settings
+	nextConsumer consumer.Metrics
 
-	config *Config     //nolint:unused
-	logger *zap.Logger //nolint:unused
+	config *Config
+	server *http.Server
 }
 
-// nolint
-func (prwc *prometheusRemoteWriteReceiver) Start(_ context.Context, _ component.Host) error {
-	panic("need implement")
+func (prw *prometheusRemoteWriteReceiver) Start(ctx context.Context, host component.Host) error {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/api/v1/write", prw.handlePRWv2)
```

Comment:
Is `/api/v1/write` really the path for the v2 PRW api?

Reply:
Yes: https://prometheus.io/docs/prometheus/latest/querying/api/#remote-write-receiver

It the same endpoint for both actually, the handler is chosen based on content negotiation.